### ISSUE TITLE
Recording persistence error in Crashlytics 

### ISSUE THREAD

**Issue by vaIerika** [GoodRequest/GoodPersistence#5]
@andrej-jasso @sebastianmraz Did you consider to add functionality to handle errors?

https://github.com/GoodRequest/GoodPersistence/blob/795727bb383b92645b7e18b4624adee36a19143e/Sources/GoodPersistence/UserDefaultsWrapper.swift#L49

**Comment by andrej-jasso** [comment]
When does this issue happen. Is this a migration issue?

**Comment by vaIerika** [comment]
Yes, it could happen during migration, as well as local changing models in some way (remaining parameters or changing type during refactor) without realizing that the object is already cached for some users. It will be really hard to trace that error without logging the error. 

Also Dominik mentioned that we added Logger here because of SLA bug and it helped to identify the problem caused by the codable inside the property.

**Comment by andrej-jasso** [comment]
So what we could add is a completion for error handling. 
Would it be an adequate solution for Notino to add errors in the publisher ?
Like in the keychain publisher<Value, Error> also for userdefaults
Do we also need a property wrapper parameter with a function time for the completion <Result> ?
